packages/token/src/erc20/interface.cairo

```diff
@@ -3,73 +3,164 @@
     fn balance_of(self: @TState, account: ContractAddress) -> u256;
+
+    /// Returns the remaining number of tokens that `spender` will be allowed to spend on behalf of
+    /// `owner` through `transfer_from`.
     fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+
+    /// Moves `amount` tokens from the caller's account to `recipient`.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+
+    /// Moves `amount` tokens from `sender` to `recipient` using the allowance mechanism.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn transfer_from(
         ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256,
     ) -> bool;
+
+    /// Sets `amount` as the allowance of `spender` over the caller's tokens.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 
+/// Interface for the optional metadata functions.
 #[starknet::interface]
 pub trait IERC20Metadata<TState> {
+    /// Returns the name of the token.
     fn name(self: @TState) -> ByteArray;
+
+    /// Returns the symbol of the token.
     fn symbol(self: @TState) -> ByteArray;
+
+    /// Returns the number of decimals used to get its user representation.
     fn decimals(self: @TState) -> u8;
 }
 
+/// Interface for the camelCase version of ERC20 functions.
 #[starknet::interface]
 pub trait IERC20Camel<TState> {
+    /// Returns the total supply of tokens.
     fn totalSupply(self: @TState) -> u256;
+
+    /// Returns the amount of tokens owned by `account`.
     fn balanceOf(self: @TState, account: ContractAddress) -> u256;
+
+    /// Returns the remaining number of tokens that `spender` will be allowed to spend on behalf of
+    /// `owner` through `transferFrom`.
     fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+
+    /// Moves `amount` tokens from the caller's account to `recipient`.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+
+    /// Moves `amount` tokens from `sender` to `recipient` using the allowance mechanism.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn transferFrom(
         ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256,
     ) -> bool;
+
+    /// Sets `amount` as the allowance of `spender` over the caller's tokens.
+    ///
+    /// Returns a boolean value indicating whether the operation succeeded.
     fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 
+/// Interface for the camelCase version of ERC20 functions without the approve function.
```

Comment:
Same applies here, and we also have to remove individual function comments. Let's always keep consistency in mind.